src/beyond_bets/transforms/vip.py

```diff
@@ -0,0 +1,31 @@
+from datetime import datetime, timedelta
+from pyspark.sql import DataFrame, SparkSession
+from beyond_bets.base.transform import Transform
+from beyond_bets.datasets.bets import Bets
+from beyond_bets.transforms.market_analyser import MarketAnalyser
+from pyspark.sql import functions as fn
+from pyspark.sql import Window
+
+
+spark = SparkSession.builder.getOrCreate()
+
+
+class TopPlayers(Transform):
+    def __init__(self):
+        super().__init__()
+        self._name: str = "TopPlayers"
+        self._inputs = {"bets": Bets()}
+
+    def _transformation(self, **kwargs: dict[str, any]) -> DataFrame:
+        window_spec = Window.orderBy(fn.col("total_bets").desc())
+        total_spend_df = MarketAnalyser(
+            entity="player_id", time_scope="day"
+        ).result()
+        total_spend_df = (
+            total_spend_df.filter(
+                fn.col("date") > (datetime.now() - timedelta(days=7))
+            )
+            .withColumn("percent_rank", fn.percent_rank().over(window_spec))
```

Comment:
using percentage rank here and understand that if multiple players have the same spending at the boundary of 1%, they will all be included. This this is fair. But the other option would be to select by total number of rows.